src/codecs/ico/decoder.rs

```diff
@@ -45,14 +45,14 @@ impl fmt::Display for DecoderError {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
             DecoderError::NoEntries => f.write_str("ICO directory contains no image"),
-            DecoderError::IcoEntryTooManyPlanesOrHotspot => {
-                f.write_str("ICO image entry has too many color planes or too large hotspot value")
-            }
-            DecoderError::IcoEntryTooManyBitsPerPixelOrHotspot => f.write_str(
-                "ICO image entry has too many bits per pixel or too large hotspot value",
-            ),
-            DecoderError::PngShorterThanHeader => {
-                f.write_str("Entry specified a length that is shorter than PNG header!")
+            DecoderError::IcoEntryTooManyPlanesOrHotspot(num_color_planes) => {
+                f.write_fmt(format_args!("ICO image entry specifies {}, which exceeds the allowed number of color planes or has an invalid hotspot value", num_color_planes))
```

Comment:
This message is unclear whether the number indicated is how many planes are in the image, or how many hotspots are

Reply:
Thank you.
I tried to improve it. 
Now users at least can know, depending on which file he or she works on.